e2e/utils/zetacore.go

```diff
@@ -81,7 +81,15 @@ func WaitCctxsMinedByInboundHash(
 		var cctxs []*crosschaintypes.CrossChainTx
 
 		timedOut := time.Since(startTime) > timeout
-		require.False(t, timedOut, "waiting cctx timeout, cctx not mined, inbound hash: %s", inboundHash)
+		if timedOut {
+			require.False(t, timedOut, "waiting cctx timeout, cctx not mined, inbound hash: %s", inboundHash)
```

Comment:
when `timedOut == true` does the program execution terminate right on this line, before `return cctxs`?

Reply:
No execution does not stop, that was why I wanted to add the return.

I think we should actually `panic()` in the `func (r *E2ERunner) FailNow() {` to halt execution of the test immediately.

Reply:
Agree, fail the test fast will be better in future. I'm was curious.

Reply:
Updated to panic in d11c53476 so that the test exits faster.